source/agora/consensus/EnrollmentManager.d

```diff
@@ -200,8 +201,9 @@ public class EnrollmentManager
 
     ***************************************************************************/
 
-    public bool addEnrollment (const ref Enrollment enroll, Height block_height,
-        scope UTXOFinder finder) @safe nothrow
+    public bool addEnrollment (const ref Enrollment enroll,
+        const ref PublicKey pubkey, Height block_height, scope UTXOFinder finder)
+        @safe nothrow
```

Comment:
Why do we need to pass the public key if we look up the UTXO ?

Reply:
Okay, looking at the rest of the code, it might make sense to just pass the `UTXO` and remove the `finder` and `pubkey` argument.

Reply:
OK, I understand you meant we can remove also the `finder` argument on `isInvalidReason`. Right?
In fact, the `finder` is used for `isInvalidReason`  in `add` of `EnrollmentPool`.

Reply:
Ah, I see, so it will be a bit complicated. Let's leave it for another day then.